### PR TITLE
Add Kafka TLS support to the producer.

### PR DESCRIPTION
**Pull request by srikartati**
Also, enhance the input consumption when initializing the
Kafka producer.

### CODE REVIEW

File: pkg/producer/kafka.go [hunk 1]
```diff
@@ -26,39 +31,67 @@ import (
...
+	if input.KafkaTLSEnabled {
+		if tlsConfig := setupTLSConfig(input.KafkaCAFile, input.KafkaTLSCertFile, input.KafkaTLSKeyFile); tlsConfig != nil {
+			kafkaConfig.Net.TLS.Config = tlsConfig
+			kafkaConfig.Net.TLS.Enable = true
+			if input.KafkaTLSSkipVerify {
+				klog.Info("kafka client TLS enabled (server certificate didn't validate)")
+			} else {
+				klog.Info("kafka client TLS enabled")
+			}
+		}
```
Comment (zyiou) [inline pkg/producer/kafka.go:83]: I can see that errors are already handled at `setupTLSConfig` method. Then is it possible that `tlsConfig` is nil? if so, we may also need to have error handling here.

File: pkg/producer/kafka.go [hunk 2]
```diff
@@ -26,39 +31,67 @@ import (
...
+	kafkaConfig.Producer.Return.Successes = input.KafkaLogSuccesses
+	kafkaConfig.Producer.Return.Errors = input.KafkaLogErrors
 
-	asyncProducer, err := sarama.NewAsyncProducer(addrs, kafkaConfig)
+	// Initialize TLS certificates
+	if input.KafkaTLSEnabled {
+		if tlsConfig := setupTLSConfig(input.KafkaCAFile, input.KafkaTLSCertFile, input.KafkaTLSKeyFile); tlsConfig != nil {
+			kafkaConfig.Net.TLS.Config = tlsConfig
+			kafkaConfig.Net.TLS.Enable = true
+			if input.KafkaTLSSkipVerify {
```
Comment (zyiou) [inline pkg/producer/kafka.go:78]: Seems that `input.KafkaTLSSkipVerify` is not used other places other than here. Do we need to handle the case when `input.KafkaTLSSkipVerify` is set?

File: pkg/producer/kafka.go [hunk 3]
```diff
@@ -26,39 +31,67 @@ import (
...
+				klog.Info("kafka client TLS enabled (server certificate didn't validate)")
+			} else {
+				klog.Info("kafka client TLS enabled")
+			}
+		}
+	}
+	if input.EnableSaramaDebugLog {
+		sarama.Logger = log.New(os.Stderr, "[sarama] ", log.LstdFlags)
+	}
+	asyncProducer, err := sarama.NewAsyncProducer(input.KafkaBrokers, kafkaConfig)
```
Comment (zyiou) [inline pkg/producer/kafka.go:88]: As long as we put TLS related config into kafkaConfig, Kafka producer will create the tls connection, right?

File: pkg/producer/kafka.go (hunk 1 above)
Comment (srikartati) [reply pkg/producer/kafka.go]: Agree. Changed the error handling, so it can be propagated to the caller.

File: pkg/producer/kafka.go (hunk 2 above)
Comment (zyiou) [reply pkg/producer/kafka.go]: Got it. We can add code to ignore secure checking with proper reasoning description like:
```
// #nosec G402: only applicable for testing purpose
t.InsecureSkipVerify = skipVerify
```

**Review by zyiou — DISMISSED** [review]
Are we planning to add tests for this PR? Otherwise LGTM overall.

**Comment by srikartati** [comment]
> Are we planning to add tests for this PR? Otherwise LGTM overall.

I do not have tests for InitKafkaProducer function. I think TLS certificates warrant some unit tests. Will add those.

File: pkg/producer/kafka.go (hunk 2 above)
Comment (shihhaoli) [reply pkg/producer/kafka.go]: Why didn't we pass input.KafkaTLSSkipVerify to setupTLSConfig(), which returns a tls.Config object?
I thought that value can be directly assigned to the returning tls.Config, such as 
tls.Config{InsecureSkipVerify: input.KafkaTLSSkipVerify} ?

Comment (srikartati) [reply pkg/producer/kafka.go]: golang lint test fails if we directly the value like above. However, I moved the assignment to `setupTLSConfig`.

Sorry for late response. I had put this PR on hold. Please take a look again.

File: pkg/producer/kafka_test.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 VMware, Inc.
```
Comment (zyiou) [inline pkg/producer/kafka_test.go:1]: 2020 -> 2021

File: pkg/test/certs.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
+package test
```
Comment (zyiou) [inline pkg/test/certs.go:1]: missing license header in this file

File: pkg/collector/process_test.go [hunk 6]
```diff
@@ -492,11 +363,11 @@ func TestTLSCollectingProcess(t *testing.T) {
 	collectorAddr := cp.GetAddress()
 	go func() {
 		roots := x509.NewCertPool()
-		ok := roots.AppendCertsFromPEM([]byte(fakeCACert))
+		ok := roots.AppendCertsFromPEM([]byte(test.FakeCACert))
```
Comment (zyiou) [inline pkg/collector/process_test.go:366]: We can also replace TLS related certs like this file in `pkg/exporter/process_test.go` and `pkg/test/exporter_collector_test.go`

Comment (srikartati) [reply pkg/collector/process_test.go:366]: done

**Review by zyiou — APPROVED** [review]
Thanks a lot for the changes. LGTM

**Pull request merged**